### PR TITLE
vite+vitest setup

### PR DESCRIPTION
**Pull request by 3cp**
Default app is moved to vite+vitest setup.
Default plugin is still with webpack, as there is some trouble for vite with plugin right now. See comments in `before.js`.

### CODE REVIEW

**Comment by 3cp** [comment]
Can be tested with

    npx makes aurelia/new#vite

**Comment by 3cp** [comment]
I cannot reproduce the same vite issue on windows 11. The CI uses windows server 2022, which is closer to windows 10 than windows 11.

However I do encounter an issue with vitest on windows 11. The vitest failed with:
```
> npm test
...
Error: ENOENT: no such file or directory, open 'C:\src\my-app.html'
```

cc @bigopon, please have a play with vitest on windows 11.

**Comment by bigopon** [comment]
Thanks @3cp , nice work on the ability to makes branch. Will test it soon when I get back to my computer.

**Comment by 3cp** [comment]
What I do know for now:

Vite touched imports in the output js of our vite conventions plugin. It changed
```js
import { customElement } from "aurelia_runtime-html";
import * as __au2ViewDef from "./my-app.html";
```
To
```js
import { customElement } from "/node_modules/.vite/deps/@aurelia_runtime-html.js?v=aa10f352";
import * as __au2ViewDef from "/src/my-app.html?import";
```

Then the vite/vitest dev server serves those contents through "root" folder.
The root is defined in vitest.config test.root. No explicit root is needed for vite.config, I guess that's because of browser env.

It seems only vitest+windows has issue with our "/src/my-app.html?import", note the `my-app.html?import` is the compiled js code, you can find it in browser dev console.

Note it doesn't have issue with "/src/my-app.ts" (the import in my-app.spec.ts which happens before the html import). Would be something with our vite plugin, but only on windows :-(

**Comment by 3cp** [comment]
vite-plugin PR created for the above issue.

**Comment by bigopon** [comment]
It seems to work fine with latest dev version (that includes the vite plugin fix)
```bash
# Next time, you can try to create similar project in silent mode:
npx makes aurelia new-project-name -s dev,typescript,vitest
```

```cmd
 ✓ test/my-app.spec.ts (1)
   ✓ my-app (1)
     ✓ should render message

 Test Files  1 passed (1)
      Tests  1 passed (1)
   Start at  23:22:06
   Duration  10.44s (transform 121ms, setup 0ms, collect 354ms, tests 39ms, environment 8.65s, prepare 850ms)
```

File: webpack/webpack.config.js [hunk 1]
```diff
@@ -237,36 +223,12 @@ module.exports = function(env, { /* @if jasmine || mocha*/runTest, /* @endif */a
               defaultShadowOptions: { mode: 'open' }
             }
           },
-          exclude: /node_modules/
-        }
-        // @endif
-        // @if css-module
```
Comment (bigopon) [inline webpack/webpack.config.js:243]: we dont need this anymore?

File: webpack/webpack.config.js [hunk 2]
```diff
@@ -13,20 +13,6 @@ const WebpackShellPluginNext = require('webpack-shell-plugin-next')
 const nodeExternals = require('webpack-node-externals');
 // @endif
 
-// @if !css-module
-const cssLoader = 'css-loader';
```
Comment (bigopon) [inline webpack/webpack.config.js:17]: 👍

**Review by bigopon — APPROVED** [review]
nice work @3cp 👍

**Comment by 3cp** [comment]
Have to wait for another "latest" release, then retest the ci.

Not sure about the vite error on the windows server 2022.

File: webpack/webpack.config.js (hunk 1 above)
Comment (3cp) [reply webpack/webpack.config.js:243]: No. css-module now follows the foo.module.css convention. Webpack css-loader supports it by default, no more explicit options to the loader.

Comment (3cp) [reply webpack/webpack.config.js:243]: Note `exclude: /node_modules/` is not removed, it's on line 267. That's the funny way git shows the diff.

**Comment by 3cp** [comment]
GitHub CI still failed Windows Server 2022 with an error I cannot reproduce on Windows 11.

**Comment by bigopon** [comment]
From the look of the stack trace, it's not clear at all how/why our plugin influenced the error. Will it have the same issue if we just use vite without our plugin at all? Is that testable?

**Comment by 3cp** [comment]
Not sure if you can replicate the error with a simple vite app without au2.

**Comment by 3cp** [comment]
I have not tried, sounds doable.

**Comment by 3cp** [comment]
I cannot reproduce the vite + Windows server 2022 error locally on Windows 11 :-(

error during build:
[vite:build-html] The "fileName" or "name" properties of emitted chunks and assets must be strings that are neither absolute nor relative paths, received "../../../../../../runneradmin/AppData/Local/Temp/test-skeletons/app-latest-vite-typescript-css-no-unit-tests-playwright-app-min-browser-evergreen/index.html".

**Comment by 3cp** [comment]
The windows problem might be something to do with mklink in windows, some folder was a "symlink" in that windows server 2022.

I will try different e2e folder.

**Comment by bigopon** [comment]
Nice @3cp 🎉

**Pull request merged**